src/compile/axis/properties.ts

```diff
@@ -167,58 +169,111 @@ export function getLabelAngle(
   }
 }
 
+export function normalizeAngleExpr(angle: SignalRef) {
+  return `(((${angle.signal} % 360) + 360) % 360)`;
```

Comment:
Should parenthesize `${angle.signal}` to preserve evaluation order

e.g., 1 + 359 % 360 !== (1 + 359) % 360

Reply:
Good catch.

Reply:
Let's deal with this separately in https://github.com/vega/vega-lite/issues/6374.